### PR TITLE
Feature/MAUI#86

### PR DESCRIPTION
**Pull request by BrandonStreets-Mcleod**
Resolves https://github.com/Software-Engineering-Red/MAUI-APP/issues/86

Acceptance criteria:

- [x] Details of a team member's current system access privileges can be viewed
- [x] Requests for lower privilege levels are automatically approved
- [x] Requests for higher privileges must be approved by the Deputy Team Leader

### CODE REVIEW

File: AppShell.xaml [hunk 1]
```diff
@@ -22,6 +22,14 @@
             Title="Rotas"
             ContentTemplate="{DataTemplate views:RotaPage}"/>
 
+        <ShellContent
+            Title="Team Members"
+            ContentTemplate="{DataTemplate views:TeamMemberPage}"/>
+
+        <ShellContent
+            Title="Privledge Requests"
```
[the remark this answers is not in the document]
Comment (BrandonStreets-Mcleod) [reply AppShell.xaml]: Are you asking if there should be access to the privilege requests page all the time or whether the page should be accessible at all? can you clarify please

File: AppShell.xaml [hunk 2]
```diff
@@ -22,6 +22,14 @@
...
+            ContentTemplate="{DataTemplate views:PrivledgeRequestsPage}"/>
```
[the remark this answers is not in the document]
Comment (BrandonStreets-Mcleod) [reply AppShell.xaml]: Same as above

File: Models/PrivledgeRequest.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace UndacApp.Models
+{
+    public class PrivledgeRequest : INotifyPropertyChanged
```
[the remark this answers is not in the document]
Comment (BrandonStreets-Mcleod) [reply Models/PrivledgeRequest.cs]: Yeah agreed, just forgot how to spell clearly. Updated now so should be all good

File: Models/PrivledgeRequest.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        public string RequestType
+        {
+            get => requestType;
+            set => SetProperty(ref requestType, value);
+        }
+
+        public string PrivledgeLevel
+        {
+            get => privledgeLevel;
+            set => SetProperty(ref privledgeLevel, value);
```
[the remark this answers is not in the document]
Comment (BrandonStreets-Mcleod) [reply Models/PrivledgeRequest.cs]: Yeah thats a good point, would you like this updated in this PR or is it good for the current state?

**Comment by BrandonStreets-Mcleod** [comment]
image
This is what it looks like when it renders but cant work it why it doesn't consistently do it

**Comment by gjemiolo-git** [comment]
Great changes!

I had a look and:
- It seems like the UI rendered before the async method was able to retrieve the tables from the DB.
- Another thing is, (I don't know why) but it seemed like if buttons were located below the `Listview`, the entries didn't want to render (Could be related to previous point with rendering the buttons introducing delay sufficient for async method to retrieve the tables?)

Overall pushed a diff with changes that seem to resolve it on my PC, could you have a look and see if it fixes it in your machine as well, hope it helps!

File: Models/PrivledgeRequest.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (gjemiolo-git) [reply Models/PrivledgeRequest.cs]: Up to you tbf. it's more QoL stuff; so not really needed

File: AppShell.xaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gjemiolo-git) [reply AppShell.xaml]: I mean the spelling 😅

File: AppShell.xaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gjemiolo-git) [reply AppShell.xaml]: ^

File: Models/PrivledgeRequest.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (gjemiolo-git) [reply Models/PrivledgeRequest.cs]: We shouldn't need it tbf; it's more QoL stuff, in the future we could make an issue and tag as a Wishlist 🤣

File: Views/PrivilegeRequestsPage.xaml.cs [hunk 5]
```diff
@@ -28,21 +28,23 @@ private async Task LoadRequests()
...
         }
         else
         {
             var selectedRequest = ltv_privilegeRequests.SelectedItem as PrivilegeRequest;
-            int updatedID = selectedRequest.ID;
-            TeamMember updatedMember = memberService.GetTeamMemberById(updatedID).Result;
-            updatedMember.AccessPrivilegeLevel = selectedRequest.PrivilegeLevel;
-            memberService.UpdateTeamMember(updatedMember);
+            int updatedID = selectedRequest.MemberID;
+            var teamMembers = new ObservableCollection<TeamMember>(await memberService.GetTeamMemberList());
```
Comment (gjemiolo-git) [inline Views/PrivilegeRequestsPage.xaml.cs:42]: Nice solution here with
```
var teamMembers = new ObservableCollection<TeamMember>(await memberService.GetTeamMemberList());
var teamMember = teamMembers.FirstOrDefault(x => x.ID == updatedID);
 ```
Alternatively `await memberService.GetTeamMemberById(updatedID);` should also work

File: Views/PrivilegeRequestsPage.xaml.cs [hunk 6]
```diff
@@ -57,6 +57,8 @@ private async void DeniedButton_Clicked(object sender, EventArgs e)
         }
         else
         {
+            var selectedRequest = ltv_privilegeRequests.SelectedItem as PrivilegeRequest;
+            await requestService.DeleteRequest(selectedRequest);
```
Comment (gjemiolo-git) [inline Views/PrivilegeRequestsPage.xaml.cs:61]: could also add here:

```
requests.Remove(selectedRequest);
ltv_privilegeRequests.SelectedItem = null;
```

So the display that sources from requests will also get updated

**Review by gjemiolo-git — APPROVED** [review]
LGTM ✅, the comments were addressed and the changes meet Acceptance criteria

**Pull request merged**